### PR TITLE
feat(optimizer/hybrid_zero_optim.py): overlap gradients last bucket allreduce and compute norm

### PR DESCRIPTION
**Pull request by huangting4201**
## Motivation

Overlap gradients last bucket allreduce communication and compute norm.

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.

### CODE REVIEW

File: internlm/model/norm.py [hunk 1]
```diff
@@ -7,23 +7,25 @@
 from torch.nn.parameter import Parameter
 
 
-def manual_rms_norm(input, normalized_shape, weight, eps):
```
Comment (sunpengsdu) [inline internlm/model/norm.py:10]: make CI happy

File: internlm/solver/optimizer/hybrid_zero_optim.py [hunk 2]
```diff
@@ -467,6 +468,37 @@ def backward(self, loss, retain_graph=False):
...
+        group_id: int = 0,
+        last_bucket: bool = False,
+        last_stage: bool = False,
+        previous_norm=None,
+    ):
+        # compute norm for gradients that have been reduced
+        params, _ = self._param_store.get_reduced_param_for_compute_norm(last_bucket=last_bucket)
+        params_list = []
+        grads_list = []
+        for param in params:
```
Comment (sunpengsdu) [inline internlm/solver/optimizer/hybrid_zero_optim.py:482]: 这样写的话，每个group都要for训练判断下，这个CPU判断会比较耗时

Comment (huangting4201) [reply internlm/solver/optimizer/hybrid_zero_optim.py:482]: 感觉1比较好，我改成1这样吧

Comment (huangting4201) [reply internlm/solver/optimizer/hybrid_zero_optim.py]: 已更新
